### PR TITLE
Add documentation of Minimum Supported Rust Version.

### PR DESCRIPTION
**Pull request by daira**
Signed-off-by: Daira Hopwood <daira@jacaranda.org>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,10 @@
 
 ## [Documentation](https://docs.rs/halo2)
 
+This crate currently has a Minimum Supported Rust Version of 1.49.0. When const generics
```
Comment (str4d) [inline README.md:7]: If we _are_ deciding to pin a MSRV now, then we should both make this change in CI, and add a `rust-toolchain` file.

**Comment by ebfull** [comment]
Can we punt this to `1.51.0`'s release?

**Comment by str4d** [comment]
I'm fine with waiting until 1.51.0, which should be released the week of March 22nd.

**Comment by daira** [comment]
Currently we fail to build on Rust 1.48.0 or earlier. This wasted several hours of my time, trying to fix compiler errors that were actually compiler bugs. So I think this should be merged now for documentation; we can do the other stuff with CI later.

**Comment by ebfull** [comment]
Okay, @daira. Sounds reasonable.

**Comment by daira** [comment]
Thanks :-)

**Pull request merged**